Tests/CoreTests.swift

```diff
@@ -126,12 +142,148 @@ class CoreTests: QuickSpec {
+                    core.destroy()
+                    
+                    expect(didCallWillDestroy).toEventually(beTrue())
+                }
+
+                it("Should trigger didDestroy") {
+                    var didCallDidDestroy = false
+
+                    core.on(InternalEvent.willDestroy.rawValue) { _ in
+                        didCallDidDestroy = true
+                    }
+                    core.destroy()
+
+                    expect(didCallDidDestroy).toEventually(beTrue())
+                }
+
+                it("Should remove listeners") {
+                    var didTriggerEvent = false
+                    let eventName = "teste"
+
+                    core.listenTo(core, eventName: eventName) { _ in
+                        didTriggerEvent = true
+                    }
+                    core.trigger(eventName)
+
+                    expect(didTriggerEvent).toEventually(beTrue())
+
+                    didTriggerEvent = false
+                    core.destroy()
+                    core.trigger(eventName)
+
+                    expect(didTriggerEvent).toEventually(beFalse())
+                }
+
+                it("Should remove all containers") {
+                    waitUntil { done in
+                        core.containers.first!.on(InternalEvent.didDestroy.rawValue) { _ in
+                            done()
+                        }
+                        core.destroy()
+                        expect(core.containers.count) == 0
+                    }
+                }
+            }
+
             context("Containers") {
                 it("Should be created given a source") {
                     expect(core.activeContainer).toNot(beNil())
                     expect(core.plugins).to(beEmpty())
                     expect(core.containers).toNot(beEmpty())
                 }
+
+                it("Should trigger willChangeActiveContainer event") {
+                    let core = Core()
+                    var didCallActiveContainerEvent = false
+
+                    core.on(InternalEvent.willChangeActiveContainer.rawValue)   { userInfo in
+                        didCallActiveContainerEvent = true
+                    }
+                    core.activeContainer = Container()
+
+                    expect(didCallActiveContainerEvent).toEventually(beTrue())
+                }
+
+                it("Should trigger didChangeActiveContainer event") {
+                    let core = Core()
+                    var didCallChangeActiveContainer = false
+
+                    core.on(InternalEvent.didChangeActiveContainer.rawValue)   { userInfo in
+                        didCallChangeActiveContainer = true
+                    }
+                    core.activeContainer = Container()
+
+                    expect(didCallChangeActiveContainer).toEventually(beTrue())
+                }
+
+                it("Should listen willChangePlayback and trigger willChangeActivePlayback for new Container") {
```

Comment:
This test might give a false positive. Imagine the scenario where no `willChangePlayback` is triggered, but instead, two `willChangeActivePlayback` was triggered. The test will pass, but the behaviour is incorrect.

We should check if every callback was called, not the number of times.

Reply:
I remove the trigger test, cuz doesnt make sense to test this action here.